### PR TITLE
Kyma 2.8.4

### PR DESCRIPTION
**Pull request by ognyvrac**
**Description**
Update Kyma to version 2.8.4. This upgrade has a breaking change - we are replacing the `istio-init` initContainer with the [Istio CNI Plugin](https://istio.io/latest/docs/setup/additional-setup/cni/). This is useful to remove our PSPs laters but would require more resources as there would be a DaemonSet of `istio-cni-node`. Overall, this would strengthen our in-cluster security as previously `istio-init` required elevated privileges.

Some tweaks to the initContainers were required and they were done in https://github.com/kyma-incubator/compass/pull/3374.

This release also updates the Istio version to 1.15.3. Previously, we had to find Istio ARM images, but with Istio 1.15 ARM is supported with the official images. We are going to use the official images as the Kyma ones are only for AMD.

Changes proposed in this pull request:
- Remove unused lines
- Update provision and benchmark scripts' Kyma CLI version
- Update docs

**Pull Request status**

- [x] Implementation
- [ ] Unit tests
- [ ] Integration tests
- [ ] `chart/compass/values.yaml` is updated 
- [ ] Mocks are regenerated, using the automated script

### CODE REVIEW

**Review by ekaterina-mitova — APPROVED** [review]
Looks good to me.

**Review by PetarTodorovv — APPROVED** [review]

File: installation/resources/kyma/kyma-overrides-minimal.yaml [hunk 1]
```diff
@@ -30,52 +25,32 @@ helm-broker:
...
+        containerRegistryPath: "docker.io"
+
   components:
     egressGateways:
       enabled: false
     ingressGateways:
-      config:
-        affinity:
-          nodeAffinity:
-            preferredDuringSchedulingIgnoredDuringExecution:
```
[the remark this answers is not in the document]
Comment (ognyvrac) [reply installation/resources/kyma/kyma-overrides-minimal.yaml:40]: I couldn't figure out why these node affinities were needed, they were introduced way back with Kyma 2.0; regarding the matching expressions - they probably didn't conflict as `requiredDuringSchedulingIgnoredDuringExecution` takes precedence and both had the same criteria.

[the remark this answers is not in the document]
Comment (PetarTodorovv) [reply installation/resources/kyma/kyma-overrides-minimal.yaml:40]: hm, then it's strange to have the `preferredDuringSchedulingIgnoredDuringExecution` when the `requiredDuringSchedulingIgnoredDuringExecution` takes precedence :D but most probably was a leftover from an old kyma version

**Pull request merged**